idaes/generic_models/unit_models/heat_exchanger.py

```diff
@@ -540,10 +538,9 @@ def initialize(
 
         init_log.info("Initialization Completed, {}".format(idaeslog.condition(res)))
         # if costing block exists, activate
-        try:
+        if hasattr(self, "costing"):
             self.costing.activate()
-        except AttributeError:
-            pass
+            costing.initialize(self.costing)
```

Comment:
Did you import the costing module above? I don't see it, so I am wondering if this is being missed by the tests as well.

Reply:
yes, I imported costing. heat_exchanger testing is correct
hx costing test (we assert that purchase cost is same after initialize and after solving the model):
    m.fs.unit.initialize()

    assert m.fs.unit.costing.purchase_cost.value == \
        pytest.approx(529738.6793, 1e-5)

    assert_units_consistent(m.fs.unit.costing)

    results = solver.solve(m)

    # Check for optimal solution
    assert results.solver.termination_condition == TerminationCondition.optimal
    assert results.solver.status == SolverStatus.ok

    assert m.fs.unit.costing.purchase_cost.value == \
        pytest.approx(529738.6793, 1e-5)